fileio/ft_read_data.m

```diff
@@ -1372,6 +1372,21 @@
       dat = read_yokogawa_data(filename, hdr, begsample, endsample, chanindx);
     end
     
+  case 'blackrock_nsx'
+    % use the NPMK toolbox for the file reading
+    ft_hastoolbox('NPMK', 1);
+    
+    % ensure that the filename contains a full path specification,
+    % otherwise the low-level function fails
+    [p,f,e] = fileparts(filename);
+    if ~isempty(p)
+      % this is OK
+    elseif isempty(p)
+      filename = which(filename);
+    end
+    orig = openNSx(filename, 'duration', [begsample endsample], 'channels', chanindx);
+    dat  = double(orig.Data);
```

Comment:
In fact, the begsample/endsample argument won't work in case of a paused file. The low-level function does not permit these arguments in that case. You can only read in the whole file then. I know that Blackrock is working on that issue too. I shot an email to the technical support there and invited him to join us here.